xbmc/pvr/windows/GUIWindowPVRRecordings.cpp

```diff
@@ -99,6 +100,20 @@ bool CGUIWindowPVRRecordingsBase::OnAction(const CAction& action)
       return true;
     }
   }
+  else if (action.GetID() == ACTION_TOGGLE_WATCHED)
+  {
+    CFileItemPtr pItem = m_vecItems->Get(m_viewControl.GetSelectedItem());
+    if (pItem->IsParentFolder())
+      return false;
+
+    if (pItem && (pItem->HasPVRRecordingInfoTag() || pItem->m_bIsFolder))
+    {
+      bool bUnwatched = pItem->GetOverlayImage() != "OverlayWatched.png";
```

Comment:
I agree this seems like a hack, and handling folders was indeed the reason I went down this route.

I was trying to avoid duplicating the logic in PVRGUIDirectory which also seems to consider subfolders, deleted items, etc. Rather than maintain this same code block in two places.

This was my best attempt based on what's currently exposed by GUIListItem.

Another approach, could be to either promote m_overlayIcon in GUIListItem to a public variable. Or add a new IsWatched() bool function to that interface which simply returns m_overlayIcon == ICON_OVERLAY_WATCHED ?

Reply:
> I was trying to avoid duplicating the logic in PVRGUIDirectory which also seems to consider subfolders, deleted items, etc. Rather than maintain this same code block in two places.

 There is always a way to refactor code so that you don't have to maintain duplicate pieces of code.

Reply:
> Another approach, could be to either promote m_overlayIcon in GUIListItem to a public variable. Or add a new IsWatched() bool function to that interface which simply returns m_overlayIcon == ICON_OVERLAY_WATCHED ?

 Also a hack.

Reply:
> Get the directory content, iterate over it and if you find the first unwatched item, break and set `bUnWatched` to true

The more I've looked at this, I think this approach makes the most sense. Any tips on how this might look would be appreciated. Not too sure how to grab the content for the currently selected directory to then iterate over.

Reply:
Sorry, no. In this form the PR will not get merged. Not knowing how to do it better does not make the approach good. 

I will take a look if I find the time, but currently I have other priorities.